T-Sanitizer error in ydb/tests/fq/http_api/test_http_api.py.TestHttpApi.test_openapi_spec

ydb/tests/fq/http_api/test_http_api.py.TestHttpApi.test_openapi_spec

**Build Information:**
- Version: bc87247b2a960f66bcf6a8a1c0afcca87c83b3fc
- Build Preset: release-tsan
- Branch: main

**Run Command:**
```
./ya make -ttt --build "release" --sanitize="thread" -DDEBUGINFO_LINES_ONLY -F 'test_http_api.py::TestHttpApi::test_openapi_spec' ydb/tests/fq/http_api
```

**Owner:** [TEAM:@ydb-platform/fq](https://github.com/orgs/ydb-platform/teams/fq)

**Test run history:** [link](https://datalens.yandex/34xnbsom67hcq?branch=main&full_name=ydb/tests/fq/http_api/test_http_api.py.TestHttpApi.test_openapi_spec&build_type=release-tsan)

More info in [dashboard](https://datalens.yandex/4un3zdm0zcnyr)

**Logs:**
- [log](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_Small_and_Medium/18732840652/ya-main-x86-64-tsan/try_3/artifacts/logs/ydb/tests/fq/http_api/test-results/py3test/testing_out_stuff/test_http_api.py.TestHttpApi.test_openapi_spec.log)
- [logsdir](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_Small_and_Medium/18732840652/ya-main-x86-64-tsan/try_3/test_artifacts/ydb-tests-fq-http_api.zip)

**Error Summary:**
```
teardown failed:
ydb/tests/fq/http_api/test_base.py:24: in teardown_class
    cls.streaming_over_kikimr.stop()
ydb/tests/tools/fq_runner/kikimr_runner.py:748: in stop
    tenant.stop()
ydb/tests/tools/fq_runner/kikimr_runner.py:62: in stop
    self.kikimr_cluster.stop(kill=False)
ydb/tests/library/harness/kikimr_runner.py:667: in stop
    raise daemon.SeveralDaemonErrors(saved_exceptions)
E   ydb.tests.library.harness.daemon.SeveralDaemonErrors: Daemon failed with message: Bad exit_code..
E   Process exit_code = 100.
E   Stdout file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/http_api/test-results/py3test/testing_out_stuff/test_http_api.py.TestHttpApi.test_openapi_spec/cp/node_1/stdout
E   Stderr file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/http_api/test-results/py3test/testing_out_stuff/test_http_api.py.TestHttpApi.test_openapi_spec/cp/node_1/stderr
E   Stderr content:
E   
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   warning: address range table at offset 0x880 has a premature terminator entry at offset 0x890
E   ==================
E   WARNING: ThreadSanitizer: data race (pid=1465422)
E   Write of size 8 at 0x72480000c600 by thread T17:
E   #0 __tsan_memset /-S/contrib/libs/clang20-rt/lib/tsan/rtl/tsan_interceptors_memintrinsics.cpp:35:3 (ydbd+0x18cb8ffd) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #1 basic_string_view /-S/contrib/libs/cxxsupp/libcxx/include/string_view:305:93 (ydbd+0x1d75cfe0) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #2 TBasicStringBuf /-S/util/generic/strbuf.h:158:22 (ydbd+0x1d75cfe0)
E   #3 THttpRequest /-S/ydb/library/actors/http/http.h:174:7 (ydbd+0x1d75cfe0)
E   #4 NHttp::THttpRequest::Clear() /-S/ydb/library/actors/http/http.cpp:70:16 (ydbd+0x1d75cfe0)
E   #5 NHttp::THttpParser<NHttp::THttpRequest>::Clear() /-S/ydb/library/actors/http/http.h:494:21 (ydbd+0x1d749124) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #6 CleanupRequest /-S/ydb/library/actors/http/http_proxy_incoming.cpp:53:22 (ydbd+0x1d829b3a) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #7 NHttp::TIncomingConnectionActor<NHttp::TPlainSocketImpl>::CleanupResponse(TIntrusivePtr<NHttp::THttpOutgoingResponse, TDefaultIntrusivePtrOps<NHttp::THttpOutgoingResponse>>&) /-S/ydb/library/actors/http/http_proxy_incoming.cpp:61:9 (ydbd+0x1d829b3a)
E   #8 NHttp::TIncomingConnectionActor<NHttp::TPlainSocketImpl>::FlushOutput() /-S/ydb/library/actors/http/http_proxy_incoming.cpp:326:21 (ydbd+0x1d828c8f) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #9 NHttp::TIncomingConnectionActor<NHttp::TPlainSocketImpl>::Respond(TIntrusivePtr<NHttp::THttpOutgoingResponse, TDefaultIntrusivePtrOps<NHttp::THttpOutgoingResponse>>) /-S/ydb/library/actors/http/http_proxy_incoming.cpp:309:20 (ydbd+0x1d82842e) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #10 NHttp::TIncomingConnectionActor<NHttp::TPlainSocketImpl>::HandleConnected(TAutoPtr<NActors::TEventHandle<NHttp::TEvHttpProxy::TEvHttpOutgoingResponse>, TDelete>&) /-S/ydb/library/actors/http/http_proxy_incoming.cpp:221:9 (ydbd+0x1d825263) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #11 NHttp::TIncomingConnectionActor<NHttp::TPlainSocketImpl>::StateConnected(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/http/http_proxy_incoming.cpp:389:13 (ydbd+0x1d8225e9) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #12 NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:350:17 (ydbd+0x1a481e9a) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #13 NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:267:28 (ydbd+0x1a4ea6bf) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #14 NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:455:39 (ydbd+0x1a4f0378) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #15 NActors::TExecutorThread::ProcessExecutorPool() /-S/ydb/library/actors/core/executor_thread.cpp:507:13 (ydbd+0x1a4efd2c) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #16 NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:533:9 (ydbd+0x1a4f1092) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #17 void* (anonymous namespace)::ThreadProcWrapper<ISimpleThread>(void*) /-S/util/system/thread.cpp:386:45 (ydbd+0x18f12d22) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   #18 (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:245:20 (ydbd+0x18f16730) (BuildId: 7c08f420329f8f640b7c83acb85629dbe668302a)
E   
E   Previous read of size 8 at 0x72480000c600 by thr
<TRUNCATED>
```

--- a/ydb/core/public_http/http_req.cpp
+++ b/ydb/core/public_http/http_req.cpp
@@ -217,9 +217,6 @@ namespace NKikimr::NPublicHttp {
     }
 
     void THttpRequestContext::DoResponse(TStringBuf status, TStringBuf message, TStringBuf body, TStringBuf contentType) const {
-        auto res = Request->CreateResponse(status, message, contentType, body);
-        ActorSystem->Send(Sender, new NHttp::TEvHttpProxy::TEvHttpOutgoingResponse(res));
-
         const TDuration elapsed = TInstant::Now() - StartedAt;
         LOG_INFO_S(*ActorSystem, NKikimrServices::PUBLIC_HTTP,
                 "HTTP response -> code: " << status <<
@@ -245,6 +242,9 @@ namespace NKikimr::NPublicHttp {
         group = group->GetSubgroup("path_pattern", PathPattern)->GetSubgroup("method", TString(Request->Method));
         group->GetSubgroup("code", TString(status))->GetCounter("count", true)->Inc();
         group->GetSubgroup("code", HttpCodeFamily(status))->GetCounter("count", true)->Inc();
+
+        ActorSystem->Send(Sender, new NHttp::TEvHttpProxy::TEvHttpOutgoingResponse(
+                    Request->CreateResponse(status, message, contentType, body)));
     }
 
     void THttpRequestContext::ParseHeaders(TStringBuf str) {

